bin/lib/check_reqs.js

```diff
@@ -19,15 +19,226 @@
 
 /*jshint node:true*/
 
-var Q     = require('Q');
+var Q     = require('q');
+var os    = require('os');
+var path  = require('path');
+var shell = require('shelljs');
 
-var MSBuildTools;
+var ConfigParser, MSBuildTools, Version, exec;
 try {
+    ConfigParser = require('../../template/cordova/lib/ConfigParser');
     MSBuildTools = require('../../template/cordova/lib/MSBuildTools');
+    exec = require('../../template/cordova/lib/exec');
+    Version = require('../../template/cordova/lib/Version');
 } catch (ex) {
     // If previous import fails, we're probably running this script
     // from installed platform and the module location is different.
+    ConfigParser = require('./ConfigParser');
     MSBuildTools = require('./MSBuildTools');
+    exec = require('./exec');
+    Version = require('./Version');
+}
+
+// The constant for VS2013 Upd2 PackageVersion. See MSDN for
+// reference: https://msdn.microsoft.com/en-us/library/bb164659(v=vs.120).aspx
+var VS2013_UPDATE2_RC = new Version(12, 0, 30324);
+var REQUIRED_VERSIONS = {
+    '8.0': {
+        os: '6.1',
+        msbuild: '11.0',
+        visualstudio: '11.0',
+        windowssdk: '8.0'
+    },
+    '8.1': {
+        os: '6.2',
+        msbuild: '12.0',
+        visualstudio: '12.0',
+        windowssdk: '8.1',
+        phonesdk: '8.1'
+    },
+    '10.0': {
+        os: '6.2',
```

Comment:
I've realized that there is another mistake from my side: there should be something like:
```javascript
'8.0': {
    os: '6.2', // instead of 6.1
    ...
},
'8.1': {
    os: '6.3', // instead of 6.2
    ...
},
'10.0': {
    os: '6.3', // Note that Windows 10 target is supported on Windows 7.
    ...        // This case get handled in checkOS function separately
    ...
```